rotkehlchen/api/rest.py

```diff
@@ -4675,15 +4680,50 @@ def export_history_events(
             )
 
         if len(history_events) == 0:
-            return api_response(
-                wrap_in_fail_result('No history processed in order to perform an export'),
+            return wrap_in_fail_result(
+                message='No history processed in order to perform an export',
                 status_code=HTTPStatus.CONFLICT,
             )
 
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            settings = self.rotkehlchen.get_settings(cursor)
+            currency = settings.main_currency.resolve_to_asset_with_oracles()
+
         serialized_history_events = []
         headers: dict[str, None] = {}
         for event in history_events:
-            serialized_event = event.serialize_for_csv()
+            if event.balance.usd_value != ZERO and currency != A_USD:
+                fiat_value = ZERO
+                try:
+                    # ask oracles for the rate in the given timestamp and currency
+                    price = self.get_rate_in_profit_currency(
+                        asset=event.asset,
+                        timestamp=ts_ms_to_sec(event.timestamp),
+                        currency=currency,
+                    )
+                    if price != ZERO_PRICE:
+                        fiat_value = event.balance.amount * price
+                    else:
+                        # if the price is zero, we query the fiat rate for currency against USD
+                        price = self.get_rate_in_profit_currency(
+                            asset=A_USD,
+                            timestamp=ts_ms_to_sec(event.timestamp),
+                            currency=currency,
+                        )
+                        fiat_value = event.balance.usd_value * price
```

Comment:
this doesn't make sense. If you are exporting in the native currency you don't care about USD. Having usd mixed with the data is worst than having a 0